python_ta/cfg/visitor.py

```diff
@@ -253,17 +257,24 @@ def visit_tryexcept(self, node: nodes.TryExcept) -> None:
 
         node.cfg_block = self._current_block
 
-        for child in node.body:
-            child.accept(self)
-        end_body = self._current_block
-
+        # Construct the exception handlers first
+        # Initialize a temporary block to later merge with end_body
+        self._current_block = self._current_cfg.create_block()
+        temp = self._current_block
         end_block = self._current_cfg.create_block()
 
         after_body = []
         for handler in node.handlers:
             h = self._current_cfg.create_block()
             self._current_block = h
             handler.cfg_block = h
+
+            # Iterate through all the exceptions this ExceptHandler handles, adding each one to the
+            # control boundary
+            for exception in _extract_exceptions(handler):
+                # Store both the 'Raise' and the exception class name
+                self._control_boundaries.append((node, {f"{nodes.Raise.__name__} {exception}": h}))
```

Comment:
This is the right idea, but I realized there's a subtlety if we want to handle overlapping except branches:

```python
try:
    raise ValueError
except ValueError:
    ...
except ValueError:
    ...
```

In this case, it's the first block that should be "linked", not the second. I think this will be reversed in the current implementation because of the call to `reversed` in `_visit_jump`. Please try that out.

As a general comment, it would be great to add some additional test cases for the changes you're making, as they are fairly subtle!

Reply:
Oh yeah, and have you tried a "bare except" (bad programming practice, but we should support it since it's valid Python code):

```python
try:
   ...
except:
   ...
```

That should act as a catch-all.